KEYCLOAK-11364 Revise Service Account roles page

thomasdarimont opened the pull request:
This eases configuration of service account users, like changing email, and managing groups.

<!---
Please read https://github.com/keycloak/keycloak/blob/master/CONTRIBUTING.md and follow these guidelines when contributing to Keycloak
-->

thomasdarimont committed:
KEYCLOAK-11364 Add link to service-account user in Service Account roles page

abstractj commented:
@thomasdarimont might worth to consider to add integration tests for this. But prior to that, I'd check on the dev mailing list if this is something that people would like to see included. @ssilvert can also share his thoughts on this.

ssilvert commented:
@thomasdarimont Sorry I'm so late responding to this. 

We probably do need a simple integration test to make sure it always works.  Also, I think it needs help text as it's not perfectly clear to the user what will happen when I click the link.

thomasdarimont commented:
I revised this PR a bit rebased it with the current master branch.

With this PR the Service Accounts page for a client looks like this:
![image](https://user-images.githubusercontent.com/314690/96647817-a9048080-132e-11eb-9463-e8e169dca387.png)

With this PR it looks like this:
![image](https://user-images.githubusercontent.com/314690/96647975-ec5eef00-132e-11eb-9aa9-49c797a9bc4a.png)

I added a link to the backing service-account user with a proper tooltip.
I also removed the duplicate client names from the section captions.

thomasdarimont committed:
KEYCLOAK-11364 Revise service-account page

thomasdarimont commented:
@ssilvert I just updated this PR do we still need a test to check that the link works? 
Is there already a test that I could extend, or do I need to create a new one?

stianst commented:
@thomasdarimont and @ssilvert should we close this PR and focus instead on the new admin console?

thomasdarimont commented:
@stianst I'd rather update this PR as this is IMO a small but useful change that helps new users

stianst commented:
Yeah, this time I actually read the JIRA and looked at the changes. It's a super simple fix that makes a fairly big difference, so I agree we should just get this merged.

The old admin console isn't fully tested in any case and we are replacing it relatively soon with the new admin console, so I'm perfectly fine with just skipping tests for this one.

stianst commented on themes/src/main/resources/theme/base/admin/resources/partials/client-service-account-roles.html:
```diff
@@ -7,10 +7,23 @@
+        <div class="form-group">
+            <label class="col-md-2 control-label">{{:: 'user-name' | translate}}</label>
+            <div class="col-md-6">
+                <a class="form-control" ng-href="#/realms/{{realm.realm}}/users/{{user.id}}" target="_blank">{{user.username}}</a>
```
I understand why you're doing _blank here, but I would just open it directly. I can always go back with back button. 
Should just render as a link rather than as an input field.

thomasdarimont replied:
Good point. I just removed the target=_blank

stianst commented on themes/src/main/resources/theme/base/admin/resources/partials/client-service-account-roles.html:
```diff
@@ -7,10 +7,23 @@
+        <h2>{{:: 'service-account' | translate}}</h2>
+
+        <div class="form-group">
+            <label class="col-md-2 control-label">{{:: 'user-name' | translate}}</label>
```
Would it be better with "Service Account User" instead of username?

Or just a link "To manage details and group mappings go here"?

thomasdarimont replied:
I adopted your suggestion and Renamed the field to "Service Account User". The tooltip now says: "Username the Service Account. To manage details and group mappings click on the username."

thomasdarimont commented:
@stianst Just rebased on current master and incorporated the suggested changes.

ssilvert commented:
@stianst @thomasdarimont I tested this manually and verified it works fine.  Merging.

xianli123 commented:
Hi @thomasdarimont @ssilvert @stianst I have already seen this feature. When I go to the Users list, I can't find the "Service-account-client name" user. I think it's confusing for users. Should we add a special user named "Service-account-client name" to the users' list as shown in the screenshot below?  What do you think?

<img width="640" alt="Screen Shot 2022-03-23 at 3 30 07 PM" src="https://user-images.githubusercontent.com/53931519/159646596-74420983-5ec7-4316-81d0-171202c46bcb.png">
